__tests__/app.test.js

```diff
@@ -1,289 +1,369 @@
-                })
-        })
+describe("Error Handling General", () => {
+  test("task 3: status code 404, responds with an error message when the path doesn't exist", () => {
+    return request(app)
+      .get("/api/notanurl")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("This page doesn't exist!");
+      });
+  });
+});
+
+describe('"POST /api/reviews/:review_id/comments"', () => {
+  test("adds a new comment and responds with it ", () => {
+    const review_id = 3;
+    const newComment = {
+      username: "mallionaire",
+      body: "I like doggos :3",
+    };
+
+    return request(app)
+      .post(`/api/reviews/${review_id}/comments`)
+      .expect(201)
+      .send(newComment)
+      .then(({ body }) => {
+        expect(body.newComment).toEqual({
+          author: "mallionaire",
+          body: "I like doggos :3",
+          comment_id: 7,
+          created_at: expect.any(String),
+          review_id: 3,
+          votes: 0,
+        });
+      });
+  });
+});
+
+describe("Error Handling for POST /api/reviews/:review_id/comments", () => {
+  test("task 10: code 400, body does not contain both mandatory keys", () => {
+    const review_id = 3;
+    const newComment = {
+      username: "mallionaire",
+    };
+    return request(app)
+      .post(`/api/reviews/${review_id}/comments`)
+      .expect(400)
+      .send(newComment)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Body doesn't contain mandatory keys");
+      });
+  });
 
-        test("status code 200, responds with a changed vote count (decrease)", () => {
-            const newVote = { inc_votes: -20 };
-            return request(app)
-                .patch('/api/reviews/2')
-                .send(newVote)
-                .expect(200)
-                .then(({ body }) => {
-                    expect(body.review).toEqual({
-                        review_id: 2,
-                        title: 'Jenga',
-                        designer: 'Leslie Scott',
-                        owner: 'philippaclaire9',
-                        review_img_url:
-                            'https://www.golenbock.com/wp-content/uploads/2015/01/placeholder-user.png',
-                        review_body: 'Fiddly fun for all the family',
-                        category: 'dexterity',
-                        created_at: new Date(1610964101251).toISOString(),
-                        votes: -15
-                    })
-                })
-        })
-    })
+  test("task 10: code 400, body does not contain both mandatory keys", () => {
+    const review_id = 333333;
+    const newComment = {
+      username: "mallionaire",
+      body: "I like doggos :3",
+    };
+    return request(app)
+      .post(`/api/reviews/${review_id}/comments`)
+      .expect(404)
+      .send(newComment)
+      .then(({ body }) => {
+        expect(body.msg).toBe("You can't add a comment here because the review doesn't exist!");
+      });
+  });
```

Comment:
What response does you server give to a post request if the review id is invalid?